test/e2e/e2e_test.go

```diff
@@ -128,9 +127,32 @@ func initializeClient(t *testing.T) (clientset.Interface, coreinformers.NodeInfo
 	sharedInformerFactory.Start(stopChannel)
 	sharedInformerFactory.WaitForCacheSync(stopChannel)
 
+	waitForNodesReady(context.Background(), t, clientSet, nodeInformer)
 	return clientSet, nodeInformer, getPodsAssignedToNode, stopChannel
 }
 
+func waitForNodesReady(ctx context.Context, t *testing.T, clientSet clientset.Interface, nodeInformer coreinformers.NodeInformer) {
```

Comment:
I can absolutely put this into another PR. I originally put it here because my code would not pass the `e2e tests` without first eliminating this race condition. I'll try again without it, maybe it will work fine on the VMs in the pipeline. Either way, I'll file a new PR with this change.

Reply:
Wonderful. I will review the PR so we can merge it and have this PR to pass all the tests.